File: .examples/.scala
class Person(val name: String, private val age: Int) {
  def greet(): String = s"Hello, my name is $name and I am $age years old."
}

object Main extends App {
  def calculateYearOfBirth(currentYear: Int, age: Int): Int = currentYear - age

  val currentYear = java.time.Year.now.getValue
  val personName = "Alice"
  val personAge = 30
  val person = new Person(personName, personAge)
  val yearOfBirth = calculateYearOfBirth(currentYear, personAge)

  println(person.greet())
  println(s"${person.name} was born in $yearOfBirth.")
}

File: .examples/.java
public class Main {
  public static void main(String[] args) {
    class Person {
      String name;
      int age;

      Person(String name, int age) {
        this.name = name;
        this.age = age;
      }

      String greet() {
        return "Hello, my name is " + name + " and I am " + age + " years old.";
      }
    }

    int currentYear = java.time.Year.now().getValue();
    String personName = "Alice";
    int personAge = 30;

    Person person = new Person(personName, personAge);
    int yearOfBirth = currentYear - personAge;

    System.out.println(person.greet());
    System.out.println(person.name + " was born in " + yearOfBirth + ".");
  }
}
